includes/polling/ports.inc.php

```diff
@@ -641,6 +631,25 @@
 
             $oid_prev = $oid.'_prev';
             if (isset($port[$oid])) {
+                if ($device['uptime'] < $config['rrd']['step']) {
+                    // If the device uptime is less than the step value then this device has most likely rebooted
+                    // meaning we might have bad snmp data for *Octets
+                    if (($oid === 'ifInOctets' || $oid === 'ifOutOctets') && ($this_port[$oid] < $port[$oid])) {
+                        // We have negative value for *Octets
+                        $max_64 = '18446744073709551616';//Unused at the moment.
+                        $max_32 = '4294967296';
+                        if ($hc_enabled !== true) {
+                            if (function_exists('gmp_sub')) {
+                                // We store the new rolled version for use in rrd. DO NOT store in the DB as we need
+                                // to calculate the value again on the next poll.
+                                $this_port[$oid . '_rolled'] = (gmp_sub($max_32, $port[$oid])) + $this_port[$oid];
+                            } else {
+                                // No GMP http://uk1.php.net/manual/en/book.gmp.php function
+                                $this_port[$oid . '_rolled'] = ($max_32 - $port[$oid]) - $this_port[$oid];
```

Comment:
Corrected.

I actually had 64Bit in there originally as well when testing for another fix but it produced crazy numbers. I think that was due to the device itself so we could possibly add rollover for 64Bit and at which point GMP is better for precision.

I can drop for now if you like.